### PR TITLE
docs(toast): toast notification rfc

### PR DESCRIPTION
**Pull request by nadiia**
screen shot 2018-09-06 at 11 43 50 am

### CODE REVIEW

File: rfcs/toast-component.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import {Toast} from 'baseui/toast';
+
+export default () => {
+  return (
+    <React.Fragment>
+      <Toast
+        onClose={()=>{console.log('Toast notification was dismissed.')}}
+        autoHideDuration={5000}
+        overrides={{
+          Bosy: props => <CustomContainer>{props.children}</CustomContainer>,
```
Comment (gergelyke) [inline rfcs/toast-component.md]: `Body`?

File: rfcs/toast-component.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+* `bottomLeft` - positioned relative to the browser window
+
+## Dependencies
+
+Does this component depend on any 3rd party packages or other internal components?
+
+## Accessibility
+
+* Toast notification container element will have `role="alert"` set on it
+* When multiple alerts are displayed and positioned relative to the browser window they won't overlap but displayed in a column way
```
Comment (gergelyke) [inline rfcs/toast-component.md]: do we have a max number of Toast that can be stacked up on each other? what would happen if there would be more than the screen can fit?

File: rfcs/toast-component.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+* `$theme: theme`
+
+## KIND Constant
+
+* `info` - Generally used for messages with an informational context
+* `success` - Generally used as a confirmation of a succesful action or operation  
+* `warning` - Generally used for messages with an warning context
+* `error` - Generally used as a notification of an error happened as a result of an action or operation
+
+## PLACEMENT Constant
```
Comment (blackswanny) [inline rfcs/toast-component.md]: `Dock`?

File: rfcs/toast-component.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+* `kind: 'info' | 'success' | 'warning' | 'error'` - Optional. Defaults to 'info'
+  Defines the type of notification
+* `onClose: function` - Optional
+  A callback function called when a notification is dismissed
+* `overrides: {Root, Body, CloseButton}` - Optional
+  Overrides for presentational components. See "Presentational Components Props API" below.
+  * `[ComponentName]: ReactComponent | {props: {}, style: {}, component: ReactComponent}` - Optional
+* `placement: 'inline' | 'topLeft' | 'top' | 'topRight' | 'bottomRight' | 'bottom' | 'bottomLeft'` - Optional. Defaults to 'top'
+  Position of a toast notification relative to the browser window. If set to 'inline' a notitication is positioned according to a normal flow of the document
+* `autoHideDuration: number` - Optional. Defaults to 0
```
Comment (blackswanny) [inline rfcs/toast-component.md]: we can call just `duration`.

File: rfcs/toast-component.md (hunk 2 above)
Comment (blackswanny) [reply rfcs/toast-component.md]: i think toasts should be controlled by some management tooling, which would avoid them to overlap each other, be hidden or outside of screen

File: rfcs/toast-component.md (hunk 1 above)
Comment (schnerd) [inline rfcs/toast-component.md]: typo: Bosy -> Body

File: rfcs/toast-component.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+* `StyledCloseButton`
+* `KIND`
+* `PLACEMENT`
+
+## `Toast` API
+
+* `children: React.node | ({dismiss}) => React.node` - Required
+  Toast notification content. The children-as-function receives a `dissmiss` method that can be called to dismiss the notification and can be used as a handler for an action inside the toast content
+* `closable: boolean` - Optional. Defaults to true
+  When set to true a close button is displayed and the notification can be dismissed by a user
```
Comment (schnerd) [inline rfcs/toast-component.md]: I use `closeable` with an `e` in modal. it seems like neither is "correct", but we should probably be consistent (I have no preference)

Comment (nadiia) [reply rfcs/toast-component.md]: I'll use your version - `closeable` - it looks closer to a correct way

File: rfcs/toast-component.md (hunk 2 above)
Comment (nadiia) [reply rfcs/toast-component.md]: That's a good question. I looked through some examples, there are cases where everything that overflows the browser window (usually older/earlier notifications) is hidden. I think scrolling the notifications list would be an option. Will add this case to this section to consider for development.

File: rfcs/toast-component.md (hunk 3 above)
Comment (nadiia) [reply rfcs/toast-component.md]: Not sure what you mean by `Dock`, could you clarify?

File: rfcs/toast-component.md (hunk 4 above)
Comment (nadiia) [reply rfcs/toast-component.md]: I'd say `autoHideDuration` gives more context on its usage. But happy to consider other options or accept this one if it gets more support.

File: rfcs/toast-component.md (hunk 3 above)
Comment (nadiia) [reply rfcs/toast-component.md]: Let's keep it in sync with the prop name

**Pull request merged**